### PR TITLE
CS-1354-ResendInvitation-Employer

### PR DESCRIPTION
**Pull request by Chazzysb**
CS-1354-ResendInvitation-Employer

### CODE REVIEW

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Controllers/ApprenticeController.cs [hunk 1]
```diff
@@ -858,5 +862,19 @@ public async Task<IActionResult> DataLockRequestRestart(DataLockRequestRestartRe
...
+        [HttpGet]
+        public async Task<IActionResult> ResendEmailInvitation([FromServices] IAuthenticationService authenticationService, ResendEmailInvitationRequest request)
+        {
+            await _commitmentsApiClient.ResendApprenticeshipInvitation(request.ApprenticeshipId, new SaveDataRequest { UserInfo = authenticationService.UserInfo }, CancellationToken.None);
+
+            return RedirectToAction("ApprenticeshipDetails", new {
+                AccountHashedId = request.AccountHashedId,
+                ApprenticeshipHashedId = request.ApprenticeshipHashedId,
+                ShowNotification = true,
+                NotificationTitle = "The invitation email has been resent."
```
Comment (Pauljgraham) [inline src/SFA.DAS.EmployerCommitmentsV2.Web/Controllers/ApprenticeController.cs:876]: hopefully these last 2 params will not be needed

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Controllers/ApprenticeController.cs [hunk 2]
```diff
@@ -858,5 +862,19 @@ public async Task<IActionResult> DataLockRequestRestart(DataLockRequestRestartRe
...
+            });
```
Comment (Pauljgraham) [inline src/SFA.DAS.EmployerCommitmentsV2.Web/Controllers/ApprenticeController.cs:877]: We should be able to put 

TempData.AddFlashMessage(notification.NotificationTitle, notification.NotificationBody, ITempDataDictionaryExtensions.FlashMessageLevel.Success);

here

**Review by Pauljgraham — COMMENTED** [review]
I think we should be able to simplify the flash message. Please see below.

I'm not sure if we can actually test the new endpoint. I would have thought it would be possible

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Controllers/ApprenticeController.cs [hunk 3]
```diff
@@ -563,10 +564,13 @@ public async Task<IActionResult> ResumeApprenticeship(ResumeRequestViewModel vie
...
         [DasAuthorize(CommitmentOperation.AccessApprenticeship)]
         [Route("{apprenticeshipHashedId}/details", Name = RouteNames.ApprenticeDetail)]
-        public async Task<IActionResult> ApprenticeshipDetails(ApprenticeshipDetailsRequest request)
+        public async Task<IActionResult> ApprenticeshipDetails(ApprenticeshipDetailsRequest request, NotificationParameters notification = null)
         {
             var viewModel = await _modelMapper.Map<ApprenticeshipDetailsRequestViewModel>(request);
 
+            if (notification?.ShowNotification == true)
+                TempData.AddFlashMessage(notification.NotificationTitle, notification.NotificationBody, ITempDataDictionaryExtensions.FlashMessageLevel.Success);
+
```
[the remark this answers is not in the document]
Comment (Chazzysb) [reply src/SFA.DAS.EmployerCommitmentsV2.Web/Controllers/ApprenticeController.cs]: Moved the flash message call to the resend route

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Controllers/ApprenticeController.cs (hunk 1 above)
Comment (Chazzysb) [reply src/SFA.DAS.EmployerCommitmentsV2.Web/Controllers/ApprenticeController.cs]: removed

**Review by Pauljgraham — APPROVED** [review]

**Pull request merged**